lldb/tools/lldb-dap/Handler/AttachRequestHandler.cpp

```diff
@@ -29,6 +30,13 @@ namespace lldb_dap {
 /// Since attaching is debugger/runtime specific, the arguments for this request
 /// are not part of this specification.
 Error AttachRequestHandler::Run(const AttachRequestArguments &args) const {
+  // Initialize DAP debugger and related components if not sharing previously
+  // launched debugger.
+  std::optional<uint32_t> target_idx = args.targetIdx;
+  if (Error err = dap.InitializeDebugger(target_idx)) {
```

Comment:
This delays `InitializeDebugger` from InitializeRequestHandler to `Launch/Attach` time. Is this timing/semantics changing required? How do we know this is safe to do without causing issues?